project/Dependencies.scala

```diff
@@ -356,4 +668,68 @@ object Dependencies {
     scalaJava8Compat,
     scalaTest % Test
   )
+
+  val validateDependencies = taskKey[Unit]("Validate Lagom dependencies to ensure they are whitelisted")
+  val dependencyWhitelist = settingKey[Seq[ModuleID]]("The whitelist of dependencies")
+
+  val validateDependenciesTask: Def.Initialize[Task[Unit]] = Def.task {
+    // We validate compile dependencies to ensure that whatever we are exporting, we are exporting the right
+    // versions. We validate test dependencies to ensure that our tests run against the same versions that we are
+    // exporting
+    val compileClasspath = (managedClasspath in Compile).value
+    val testClasspath = (managedClasspath in Test).value
+    val cross = CrossVersion(scalaVersion.value, scalaBinaryVersion.value)
+    val log = streams.value.log
+    val svb = scalaBinaryVersion.value
+
+    val whitelist = dependencyWhitelist.value.map { moduleId =>
+      val crossModuleId = cross(moduleId)
+      (crossModuleId.organization, crossModuleId.name) -> crossModuleId.revision
+    }.toMap
+
+    def collectProblems(scope: String, classpath: Classpath) = {
+      classpath.collect(Function.unlift { dep =>
+        val moduleId = dep.get(moduleID.key).getOrElse {
+          sys.error(s"Managed classpath dependency without moduleID: $dep")
+        }
+
+        whitelist.get((moduleId.organization, moduleId.name)) match {
+          case None =>
+            Some(moduleId -> s"$scope dependency not in whitelist: $moduleId")
+          case Some(unmatched) if moduleId.revision != unmatched =>
+            Some(moduleId -> s"$scope dependency ${moduleId.organization}:${moduleId.name} version ${moduleId.revision} doesn't match whitelist version $unmatched")
+          case _ => None
+        }
+      })
+    }
+
+    val problems = collectProblems("Compile", compileClasspath) ++ collectProblems("Test", testClasspath)
+
+    if (problems.nonEmpty) {
+      problems.foreach(p => log.error(p._2))
+
+      log.debug {
+        // This makes it very easy to fix the problem, by outputting a formatted list of dependencies to add.
+        problems.map { problem =>
+          val mid = problem._1
+          val cross = mid.name.endsWith("_" + svb)
+          val m = if (cross) "%%" else "%"
+          val name = if (cross) mid.name.dropRight(svb.length + 1) else mid.name
+          s""""${mid.organization}" $m "$name" % "${mid.revision}""""
+        }.sorted.mkString(
+          "The following dependencies need to be added to the whitelist:\n",
+          ",\n",
+          ""
+        )
+      }
+      throw new DependencyWhitelistValidationFailed
+    }
+  }
```

Comment:
Currently the whitelist is sorted alphabetically by `groupId+artifactId` except for those dependencies listed in `libraryFamily`'s. When a new transitive dependency appears and only the version has changed, this alert may drive us into adding the offending library into the list when it may already be on the set of `libraryFamily`. 

I don't expect the message to detect the library that needs adding is on the `libraryFamily` but a fair warning to avoid tripping. How about: `"The following dependencies need to be added to the whitelist (review the both list and libraryFamilies):\n".